### PR TITLE
[#1325] Log metrics

### LINKED ISSUE #1325: Log metrics
The goal is to get a chance to export logs with the metrics. Mostly focused on the syncing metrics but could be used for any interesting logs as well. Such log would be for example `connectivity issues` or errors is general.

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #1325 

- the logger has been extended to log the level as well
- there is only partial match of levels between SDK logger levels, OSLogEntryLogLevel and OSLogType so only debug, info, error are fully matched
- this is a base for the exporter on client's side
- Scan & Enhance logs added
- checkpoints updated
- every CBP action is measured separately and collects the data, when the sync is done it dumps overview of the run to the logger
- next run clears out the previous data and starts to collect fresh reports for the run

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

**Review by Chlup — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,35 @@ All notable changes to this library will be documented in this file.
...
+
+### [#1325] Log metrics
+The sync process is measured and detailed metrics are logged for every sync run. The data are logged using os_log so any client can export it. Verbose logs are under `sdkLogs_default` category, `default` level. Sync specific logs use `error` level. 
+
+## Checkpoints
+
+Mainnet
+
+````
+Sources/ZcashLightClientKit/Resources/checkpoints/mainnet/2270000.json
```
Comment (HonzaR) [inline CHANGELOG.md:22]: Should we inform about checkpoints in the changelog? It's an SDK internal implementation, which we could mention, but we don't necessarily need to share details about it.

File: Tests/PerformanceTests/SDKMetricsTests.swift [hunk 2]
```diff
@@ -1,152 +0,0 @@
...
-//  SDKMetricsTests.swift
-//  
-//
-//  Created by Lukáš Korba on 19.12.2022.
-//
-
-import XCTest
-@testable import ZcashLightClientKit
-
-final class SDKMetricsTests: XCTestCase {
```
Comment (HonzaR) [inline Tests/PerformanceTests/SDKMetricsTests.swift:11]: Should we rework these tests to have the SDKMetrics logic still covered by tests?

**Review by HonzaR — COMMENTED** [review]
Look's good to me. I left only a few notes inline. Plus, here are general questions: 
- What is the motivation for removing enable/disable API from the metrics collecting logic? We could still have switching logging and collecting metrics in the public API.
- Could we prepare tests on ensuring expected logs are pasted into the device logger stack and another for SDKMetricks?

File: CHANGELOG.md (hunk 1 above)
Comment (LukasKorba) [reply CHANGELOG.md:22]: Pacu started this initiative in this structure so I continue. We may not need to mention it but SDK changelog is a bit more technical than Zashi one. So I didn't find it pointless.

**Pull request merged**